### ISSUE TITLE
Change annotations implementation

### ISSUE THREAD

**Issue by ajmacdonald** [sgsinclair/Voyant#352]
AnnotateIt is shutting down and no longer accepts new accounts.

**Comment by sgsinclair** [comment]
I assume @SusanBrown is following this one too.

**Comment by SusanBrown** [comment]
Thanks. I didn't know about this.

Sent from my iPhone

On Mar 16, 2017, at 10:25 AM, Stéfan Sinclair <notifications@github.com<mailto:notifications@github.com>> wrote:


I assume @SusanBrown<https://github.com/SusanBrown> is following this one too.

**Comment by SusanBrown** [comment]
Hi @ajmacdonald 

Jasmine has done some experimenting with several of the leading annotation tools and they seem to have trouble anchoring their annotations to specific strings of text in the DToC. 

The document with her notes in it is here: https://docs.google.com/document/d/1nSuEG8aoygOnYexvzfp1nyW3OX56GdH1bNsbBej4WC0/edit#

Would you be able to investigate what might be going on?

**Comment by ajmacdonald** [comment]
@SusanBrown there was a similar issue with the previous annotator. The solution at the time was to integrate the annotator within DToC so that we could fully control saving and loading of annotations.
The candidates you're looking at seem to be browser plugins, so that type of solution is not possible this time.

One possible solution is to modify the URL to add the docId when a document is loaded.
@sgsinclair I think there was an argument against this approach previously. Can you think of reasons not to do this?

**Comment by ajmacdonald** [comment]
@sgsinclair 
In the initial commit [the code to do that is commented out](https://github.com/sgsinclair/Voyant/blob/30c7b3bcfa4b2d1308946f8f6ce9e3275c0ee2c2/src/main/webapp/dtoc/dtoc.reader.js#L248-L264). Again, I can't recall why though.

**Comment by ajmacdonald** [comment]
Hypothesis is based on Annotator.js so we might be able to use similar adaptations after all.

**Comment by sgsinclair** [comment]
@ajmacdonald regarding the URL I think using doc ID is more fragile because it may change when a corpus is recreated, but docIndex (the document's position in the corpus) shouldn't (by definition a corpus with a different order is a different corpus).

**Comment by ajmacdonald** [comment]
@sgsinclair if the corpus is recreated, wouldn't it then have a different corpus ID as well?
Things get a bit tricky in DToC because we're allowing doc order to be changed through the curation mode (without it being saved as a new corpus).

**Comment by sgsinclair** [comment]
Yes, in normal Voyant world a corpus is a unique arrangement of documents in a set order. We could force a reordered corpus to use a new ID (new URL) or else we need to handle the reordering virtually, mapping the original order to the new order.

At a quick pass, where do you think order would need to be virtualized?

**Comment by ajmacdonald** [comment]
I already have some code in place that's supposed to do that (using docIndex) but apparently [it's a bit flaky](https://github.com/sgsinclair/Voyant/issues/424) so I was thinking about using docIds instead

**Comment by sgsinclair** [comment]
ID is definitely more robust, just the issue that an exported URL with ordering would be much longer. But maybe that's a separate issue with its own solution (a param that succinctly maps indices) but that's only used once
